src/openforms/contrib/objects_api/validators.py

```diff
@@ -78,3 +92,52 @@ def validate_document_type_references(
 
     if errors:
         raise ValidationError(errors)
+
+
+def validate_object_ownership(submission: Submission) -> None:
```

Comment:
I'm not sure that this file is the correct place, we have validators here for configuration validation, but not stuff for runtime submissions. I'd be inclined to just put this in its own module, `ownership_verification.py` or something like that